### PR TITLE
[Feature] Add LUT transform

### PR DESCRIPTION
**Pull request by yamengxi**

### CODE REVIEW

File: tests/test_image/test_photometric.py [hunk 1]
```diff
@@ -200,3 +200,21 @@ def _adjust_contrast(img, factor):
                 _adjust_contrast(img, factor).astype(np.int32),
                 rtol=0,
                 atol=1)
+
+    def test_lut_transform(self):
+        lut_table = np.array(list(range(256)))
+
+        img = mmcv.lut_transform(self.img, lut_table)
+        assert id(img) != id(self.img)
```
Comment (xvjiarui) [inline tests/test_image/test_photometric.py:208]: Why testing `id`?

File: mmcv/image/photometric.py [hunk 2]
```diff
@@ -224,3 +224,26 @@ def adjust_contrast(img, factor=1.):
...
+
+def lut_transform(img, lut_table):
+    """Performs a look-up table transform of an array.
+
+    The function lut_transform fills the output array with values from the
+    look-up table. Indices of the entries are taken from the input array.
+
+    Args:
+        img (ndarray): Image to be transformed.
+        lut_table (ndarray): look-up table of 256 elements; in case of
```
Comment (xvjiarui) [inline mmcv/image/photometric.py:237]: Multi channel is not tested.

File: tests/test_image/test_photometric.py (hunk 1 above)
Comment (yamengxi) [reply tests/test_image/test_photometric.py:208]: Test whether the original image is changed.

File: mmcv/image/photometric.py (hunk 2 above)
Comment (yamengxi) [reply mmcv/image/photometric.py:237]: It can support arbitrary shape ndarray.

**Review by xvjiarui — APPROVED** [review]

File: mmcv/image/photometric.py [hunk 3]
```diff
@@ -224,3 +224,26 @@ def adjust_contrast(img, factor=1.):
         img.astype(np.float32), factor, degenerated.astype(np.float32),
         1 - factor, 0)
     return contrasted_img.astype(img.dtype)
+
+
+def lut_transform(img, lut_table):
+    """Performs a look-up table transform of an array.
```
Comment (xvjiarui) [inline mmcv/image/photometric.py:230]:
```suggestion
    """Transform array by look-up table.
```

File: mmcv/image/photometric.py [hunk 4]
```diff
@@ -224,3 +224,26 @@ def adjust_contrast(img, factor=1.):
...
+            multi-channel input array, the table should either have a single
+            channel (in this case the same table is used for all channels) or
+            the same number of channels as in the input array.
+
+    Returns:
+        ndarray: The transformed image.
+    """
+    assert isinstance(img, np.ndarray)
+    assert 0 <= np.max(img) and np.max(img) <= 255
+    assert len(lut_table.shape) == 1 and lut_table.shape[0] == 256
```
Comment (hellock) [inline mmcv/image/photometric.py:247]: assert lut_table.shape == (256, )

**Pull request merged**